File: goblinframework-cache/src/main/java/org/goblinframework/cache/core/module/monitor/VMC.java
package org.goblinframework.cache.core.module.monitor;

import org.goblinframework.api.monitor.InstructionTranslator;
import org.goblinframework.core.monitor.InstructionImpl;
import org.goblinframework.core.util.StringUtils;
import org.jetbrains.annotations.NotNull;

import java.util.List;

final public class VMC extends InstructionImpl {

  public String operation;
  public List<String> keys;

  public VMC() {
    super(Id.VMC, Mode.SYN, true);
  }

  @NotNull
  @Override
  public InstructionTranslator translator() {
    return pretty -> {
      if (!pretty) {
        return asShortText();
      }
      String operation = StringUtils.defaultString(VMC.this.operation);
      if (keys == null || keys.isEmpty()) {
        return String.format("%s %s", asLongText(), operation);
      } else {
        String ks = StringUtils.join(keys, " ");
        return String.format("%s %s %s", asLongText(), operation, ks);
      }
    };
  }
}

File: goblinframework-cache/src/main/java/org/goblinframework/cache/core/module/monitor/VMC.kt
package org.goblinframework.cache.core.module.monitor

import org.goblinframework.api.monitor.Instruction
import org.goblinframework.api.monitor.InstructionTranslator
import org.goblinframework.core.monitor.InstructionImpl
import org.goblinframework.core.util.StringUtils

class VMC : InstructionImpl(Instruction.Id.VMC, Instruction.Mode.SYN, true) {

  var operation: String? = null
  var keys: List<String>? = null

  override fun translator(): InstructionTranslator {
    return InstructionTranslator { pretty ->
      if (!pretty) {
        return@InstructionTranslator asShortText()
      }
      val operation = StringUtils.defaultString(operation) { "" }
      if (keys.isNullOrEmpty()) {
        return@InstructionTranslator String.format("%s %s", asLongText(), operation)
      } else {
        val ks = StringUtils.join(keys, " ")
        return@InstructionTranslator String.format("%s %s %s", asLongText(), operation, ks)
      }
    }
  }
}
